packages/frontend/src/features/semanticViewer/components/RunSemanticQueryButton.tsx

```diff
@@ -61,14 +61,8 @@ export const RunSemanticQueryButton: FC = () => {
     useEffect(() => {
         if (!resultsColumns || !resultsData) return;
 
-        const vizColumns =
-            SemanticViewerResultsRunner.convertColumnsToVizColumns(
-                fields,
-                resultsColumns,
-            );
-
-        dispatch(setResults({ results: resultsData, columns: vizColumns }));
-    }, [dispatch, resultsData, resultsColumns, fields]);
+        dispatch(setResults({ results: resultsData, columns }));
+    }, [dispatch, resultsData, resultsColumns, fields, columns]);
```

Comment:
By using `columns` here directly from the state the results table and table viz don't work correctly:

results table:
![image](https://github.com/user-attachments/assets/d3580ea9-7491-44a4-b150-54c839c0e73f)

table viz:
![image](https://github.com/user-attachments/assets/487e4618-b416-4c46-815d-b70bac55a8ff)

this is because `state.columns` is only being set on `setResults` action, and therefore will keep being empty array.

With the previous code, where we converted the columns array that came in the response to `VizColumn` and passed that to the action the columns would always be just the ones in the results.

Reply:
Updated. I think this is fixed.